config/common.yml

```diff
@@ -42,53 +42,58 @@ services:
     arguments:
       - '@PrestaShop\Module\PrestashopCheckout\PayPal\PayPalConfiguration'
 
-  ps_checkout.tactician.bus:
-    class: 'League\Tactician\CommandBus'
-    factory: [ '@PrestaShop\Module\PrestashopCheckout\CommandBus\TacticianCommandBusFactory', "create" ]
-
   ps_checkout.bus.command:
     class: 'PrestaShop\Module\PrestashopCheckout\CommandBus\TacticianCommandBusAdapter'
     public: true
-    arguments:
-      - "@ps_checkout.tactician.bus"
+    factory: [ '@PrestaShop\Module\PrestashopCheckout\CommandBus\TacticianCommandBusFactory', "create" ]
+
+  ps_checkout.bus.query:
+    class: 'PrestaShop\Module\PrestashopCheckout\CommandBus\TacticianQueryBusAdapter'
+    public: true
+    factory: [ '@PrestaShop\Module\PrestashopCheckout\CommandBus\TacticianQueryBusFactory', "create" ]
```

Comment:
In PrestaShop V9 we removed tactician in favor of Symfony Messenger as the middleware that handles the command bus

I'm not sure forcing a custom integration of tactician in this module is a good idea, ideally you should rely on the core messenger like any other CQRS commands The migration is quite straight forward actually you just need to add this attribute on the handler class `#[AsCommandHandler]` and add `autowire` and `autoconfigure` on the handler service definition

Reply:
Oh actually I noticed that the CQRS commands are also used in FO controllers, I guess that explains the need for this custom bus right?

Reply:
Yes, I guess so. But I would need to make some tests, because this code comes from as far as PS 1.7 and probably there wasn't a command handler back then